src/GuessControl.jsx

```diff
@@ -1,47 +1,27 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import Button from "./Button";
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+const GuessControl = ({ onGuess }) => {
+  const [currentGuess, setCurrentGuess] = useState("");
 
-    this.state = {
-      currentGuess: "",
-    };
+  const handleInputChange = (event) => {
+    setCurrentGuess(event.target.value);
+  };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
-  }
+  const onSubmitGuess = () => {
+    onGuess(Number(currentGuess));
+    setCurrentGuess(" ");
```

Comment:
You should change the `" "` for an empty string `""`, otherwise you'll get the following warning message in the console:
`The specified value " " cannot be parsed, or is out of range`